main.js

```diff
@@ -1,153 +1,225 @@
-}
+function onSubmitHandler(e) {
+    e.preventDefault();
 
-// Output exeption message
-function showError(elementId, textError)
-{    
-    removeError();
-    const error = document.getElementById(elementId).appendChild(document.createElement("p"));
-    error.classList.add("error");
-    error.textContent = textError;
+    if (checkLoginPassword(formValues, userInfo)) {
+        userInfo.token = "xxx";
+        alert("correct");
+    } else {
+        setFormErrors("authentication", "Username or password is not correct");
+    }
 }
 
-// Delete exeption message
-function removeError(){
-    if(document.getElementById("form_username").contains(document.querySelector(".error"))){
-        document.getElementById("form_username").removeChild(document.querySelector(".error"));
-    } else if(document.getElementById("form_password").contains(document.querySelector(".error"))){
-        document.getElementById("form_password").removeChild(document.querySelector(".error"));
+function renderLoginForm() {
+    const form = createFormContainer(onSubmitHandler);
+    const loginForm = form.children.item(0);
+
+    loginForm.appendChild(createTitle("Login form"));
+    loginForm.appendChild(createInputField({
+        divId: "form_username", 
+        paragraphId: "username_error", 
+        paragraphClass: "error",
+        inputName: "username", 
+        labelText: "Username", 
+        inputValue: formValues.username, 
+        error: formErrors.username,
+        onClick: setTouchedElements, 
+        onInput: function(e) {
+            setFormValue(e.target.id, e.target.value);
+        }
+    }));
+    loginForm.appendChild(createInputField({
+        divId: "form_password", 
+        paragraphId: "password_error", 
+        paragraphClass: "error",
+        inputName: "password", 
+        labelText: "Password", 
+        inputValue: formValues.password, 
+        error: formErrors.password,
+        onClick: setTouchedElements, 
+        onInput: function(e) {
+            setFormValue(e.target.id, e.target.value);
+        }
+    }));
+    loginForm.appendChild(createButton("login_btn", setTouchedElements));
+    loginForm.appendChild(createParagraph("authentication_error", formErrors.authentication ,"error")); 
+
+    const currentForm = document.querySelector(".form_container");
+    if(!currentForm) {
+        document.body.append(form);
+    } else {
+        replaceChangedChild(currentForm, form);
     }
 }
 
-// Creating html registration form
-function createLoginForm() {
-    document.body.append(createFormContainer());
-
-    const parentElement = document.getElementById("login");
-    parentElement.appendChild(createTitle("Login form"));
-    parentElement.appendChild(createInputField("form_username","username", "Username"));
-    parentElement.appendChild(createInputField("form_password","password", "Password"));
-    parentElement.appendChild(createButton("login_btn")); 
+function replaceChangedChild(currentForm, form) {
```

Comment:
rename `currentForm` and `form` according to `replaceChangedChild`

Reply:
Done